src/System.Collections/src/System/Collections/ThrowHelper.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Collections
+{
+    internal static class ThrowHelper
+    {
+        internal static void ThrowInvalidOperationException_ConcurrentOperationsNotSupported() => throw CreateInvalidOperationException_ConcurrentOperationsNotSupported();
```

Comment:
This can be just:

`internal static void ThrowInvalidOperationException_ConcurrentOperationsNotSupported() => throw new InvalidOperationException(SR.InvalidOperation_ConcurrentOperationsNotSupported);`

The complex throw helper pattern with intermediate CreateInvalidOperationException is a necessary only for cases where the code needs to be compatible with old JITs that do not perform the right optimizations. It is not the case for System.Collections.

Reply:
I see. Thought it was for all cases. Thanks, will update the PR.

Reply:
This drop in the bucket since the rest of the Hashset is not using the ThrowHelper. Using it in one place only helps very little.

These manual throw helpers are pain to deal with for Mono folks who are trying to reuse the code. cc @marek-safar 

Unless this has a measurable performance impact for something that matters, we should introducing these manual throw helpers.

If we care about the performance benefits, we should build the plugin for IL linker that autogenerates them.

Reply:
> If we care about the performance benefits, we should build the plugin for IL linker that autogenerates them everywhere.

👍 I'd be very happy with that :)